test/integration/multikueue/multikueue_test.go

```diff
@@ -48,136 +48,157 @@ import (
-
-		managerMultikueueSecret2 = &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "multikueue2",
-				Namespace: managersConfigNamespace.Name,
-			},
-			Data: map[string][]byte{
-				kueuealpha.MultiKueueConfigSecretKey: w2Kubeconfig,
-			},
-		}
-		gomega.Expect(managerTestCluster.client.Create(managerTestCluster.ctx, managerMultikueueSecret2)).To(gomega.Succeed())
-
-		workerCluster1 = utiltesting.MakeMultiKueueCluster("worker1").KubeConfig(kueuealpha.SecretLocationType, managerMultikueueSecret1.Name).Obj()
-		gomega.Expect(managerTestCluster.client.Create(managerTestCluster.ctx, workerCluster1)).To(gomega.Succeed())
-
-		workerCluster2 = utiltesting.MakeMultiKueueCluster("worker2").KubeConfig(kueuealpha.SecretLocationType, managerMultikueueSecret2.Name).Obj()
-		gomega.Expect(managerTestCluster.client.Create(managerTestCluster.ctx, workerCluster2)).To(gomega.Succeed())
-
-		managerMultiKueueConfig = utiltesting.MakeMultiKueueConfig("multikueueconfig").Clusters(workerCluster1.Name, workerCluster2.Name).Obj()
-		gomega.Expect(managerTestCluster.client.Create(managerTestCluster.ctx, managerMultiKueueConfig)).Should(gomega.Succeed())
-
-		multikueueAC = utiltesting.MakeAdmissionCheck("ac1").
-			ControllerName(multikueue.ControllerName).
-			Parameters(kueuealpha.GroupVersion.Group, "MultiKueueConfig", managerMultiKueueConfig.Name).
-			Obj()
-		gomega.Expect(managerTestCluster.client.Create(managerTestCluster.ctx, multikueueAC)).Should(gomega.Succeed())
-
-		ginkgo.By("wait for check active", func() {
-			updatedAc := kueue.AdmissionCheck{}
-			acKey := client.ObjectKeyFromObject(multikueueAC)
-			gomega.Eventually(func(g gomega.Gomega) {
-				g.Expect(managerTestCluster.client.Get(managerTestCluster.ctx, acKey, &updatedAc)).To(gomega.Succeed())
-				cond := apimeta.FindStatusCondition(updatedAc.Status.Conditions, kueue.AdmissionCheckActive)
-				g.Expect(cond).NotTo(gomega.BeNil())
-				g.Expect(cond.Status).To(gomega.Equal(metav1.ConditionTrue), "Reason: %s, Message: %q", cond.Reason, cond.Message)
-			}, util.Timeout, util.Interval).Should(gomega.Succeed())
-		})
-
-		managerCq = utiltesting.MakeClusterQueue("q1").
-			AdmissionChecks(multikueueAC.Name).
-			Obj()
-		gomega.Expect(managerTestCluster.client.Create(managerTestCluster.ctx, managerCq)).Should(gomega.Succeed())
+type MultikueueTestVars struct {
+	managerNs *corev1.Namespace
+	worker1Ns *corev1.Namespace
+	worker2Ns *corev1.Namespace
+
+	managerMultikueueSecret1 *corev1.Secret
+	managerMultikueueSecret2 *corev1.Secret
+	workerCluster1           *kueuealpha.MultiKueueCluster
+	workerCluster2           *kueuealpha.MultiKueueCluster
+	managerMultiKueueConfig  *kueuealpha.MultiKueueConfig
+	multikueueAC             *kueue.AdmissionCheck
+	managerCq                *kueue.ClusterQueue
+	managerLq                *kueue.LocalQueue
+
+	worker1Cq *kueue.ClusterQueue
+	worker1Lq *kueue.LocalQueue
+
+	worker2Cq *kueue.ClusterQueue
+	worker2Lq *kueue.LocalQueue
+}
```

Comment:
I think it's batter to keep the old approach on variables , beforeEach  and afetrEach.

Even if maybe it's not the case now, in the future it's likely that the two test scenarios  (with /without GC) will need a different set of variables or different setup logic.